### PR TITLE
fix: pass through logger to @octokit/auth-app

### LINKED ISSUE #218: authStrategy does not inherit logger from Octokit
**What happened?**

`@octokit/auth-app` introduced a new `log` option, similar to [Octokit's logging](https://github.com/octokit/core.js/tree/458298288cb27ed3e7f6eaef716b7be1bc97f5cc#logging) via https://github.com/octokit/auth-app.js/pull/190. But when I pass a custom logger as `options.log` to Octokit, it is not passed down to the configured authentication strategy.

Example:

```js
const { Octokit } = require("@octokit/core")
const { createAppAuth } = require("@octokit/auth-app");

const octokit = new Octokit({
  log: myCustomLogger
  authStrategy: createAppAuth,
  auth: {
    id: 1,
    privateKey: PRIVATE_KEY,
    installationId: 2
  }
})

octokit.request('GET /repos/octokit/core.js')
```

If GitHub responds with a `401` (usually due to a replication lag), `@octokit/auth-app` is logging a warning:

> [@octokit/auth-app] Retrying after 401 response to account for token replication delay (retry: 1, wait: 1s)

But it logs using the default `console.warn`.

Test case that reproduces the problem:
https://runkit.com/gr2m/octokit-core-authstrategy-custom-logger/1.0.0

**What did you expect to happen?**

I expected that `myCustomLogger` from the `Octokit` constructor options is passed down to `createAppAuth({ log })`, so that `myCustomLogger.log.warn` is used to log the warning instead of `console.warn`.

**What the problem might be**

We do pass the internal `request` method as `options.request` to `createAppAuth(options)` by default, see 

https://github.com/octokit/core.js/blob/458298288cb27ed3e7f6eaef716b7be1bc97f5cc/src/index.ts#L141-L148

We have to do the same with the internal `log` object

**How to contribute**

Comment below if you would like to work on this issue. Then follow the instructions in our [contributing guidelines](https://github.com/octokit/core.js/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by ewanharris**
Closes #218

I reworked the runkit example to use fetch-mock and then added it to `auth.test.ts`, I also had to force an update of `@octokit/auth` to pull in the changes in `@octokit/auth-app`

### CODE REVIEW

File: test/auth.test.ts [hunk 1]
```diff
@@ -353,4 +353,46 @@ describe("Authentication", () => {
 
     expect(mock.done()).toBe(true);
   });
+
+  it("should pass through the logger", async () => {
```
Comment (gr2m) [inline test/auth.test.ts:357]: I like adding an issue number to tests so people can look up the decision process behind this functionality

```suggestion
  it("should pass through the logger (#1277)", async () => {
```

**Review by gr2m — APPROVED** [review]
I'd add a reference to the issue in the test title, other than that this is great 👍🏼

File: test/auth.test.ts (hunk 1 above)
Comment (ewanharris) [reply test/auth.test.ts]: Nice, I hadn't thought of doing that before

**Comment by gr2m** [comment]
Thanks so much Ewan, great pull request! Welcome to @octokit! Let me know if you'd like to contribute more. And if yes, what you are interested in to learn / work on :)

**Pull request merged**